main.tf

```diff
@@ -45,15 +23,14 @@ data "aws_subnet" "pod" {
 }
 
 locals {
-  # Get zones where ALL instance types are offered(intersection).
-  zone_intersection_instance_offerings = setintersection([for k, v in data.aws_ec2_instance_type_offerings.nodes : toset(v.locations)]...)
-  # Get the zones that are available and offered in the region for the instance types.
-  az_names    = var.vpc_id != null ? distinct(data.aws_subnet.private[*].availability_zone) : length(var.availability_zones) > 0 ? var.availability_zones : data.aws_availability_zones.available.names
-  offered_azs = setintersection(local.zone_intersection_instance_offerings, toset(local.az_names))
```

Comment:
There is the possibility of a mismatch between this old set of randomly selected AZs and the new, sorted set resulting in a recreation of the subnets, like this CircleCI test:

https://app.circleci.com/pipelines/github/dominodatalab/terraform-aws-eks/506/workflows/f9bd7e38-6580-41b0-bcce-0abc05461162/jobs/334

I'm not sure there's any real world usage so I'm not sure if we should bother trying to match up between the two (i.e. sort by name of the AZ instead of ID).

Reply:
Ended up pushing a6d59de for backwards compatibility, it's just easier overall to understand I think.